vars/camelwebservice_quickstart.groovy

```diff
@@ -0,0 +1,136 @@
+                    name: 'TAG',
+                    description: 'Optional. The tag from the spec repo. If empty, defaults to pulling from main branch.'
+            )
+            string(
+                    name: 'REPO_NAME',
+                    description: 'Required. A name of the new repo.'
+            )
+            string(
+                    name: 'REPO_DESC',
+                    description: 'Optional. A description of the new repository.'
+            )
+            booleanParam(
+                    name: 'IS_PRIVATE',
+                    description: 'Whether the repository is to private or not.',
+                    defaultValue: 'false',
+            )
+            string(
+                    name: 'GROUP_ID',
+                    description: 'Required. The group ID of the new project.'
+            )
+            string(
+                    name: 'ARTIFACT_ID',
+                    description: 'Required. The artifact ID (i.e. name) of the new project.'
+            )
+            string(
+                    name: 'VERSION',
+                    defaultValue: '0.0.1',
+                    description: 'Required. The version of the project.'
+            )
+        }
+        environment {
+            GIT_CREDS = credentials("${TAVROS_GIT_CREDS}")
+            GIT_HOST = "${TAVROS_GIT_HOST}"
+        }
+        stages {
+            stage('Checkout Spec Repo') {
+                steps  {
+                    script {
+                        dir("spec-files") {
+                            checkout([
+                                    $class: 'GitSCM',
+                                    branches: [[name: "${TAG}" ? "refs/tags/${TAG}" : "*/main"]],
+                                    userRemoteConfigs: [[
+                                            credentialsId: "${TAVROS_GIT_CREDS}",
+                                            url: "https://${GIT_HOST}/${ORG}/${API_REPO_NAME}.git"
+                                    ]]
+                            ])
+                        }
+                    }
+                }
+            }
+            stage('Generate Project with OpenAPI Archetype') {
+                steps {
+                    container('maven') {
+                        script {
+                            dir("spec-files") {
+                                utils.shResource "maven-archetype-generate.sh"
+                            }
+                        }
+                    }
+                }
+            }
+            stage('Create Camel Web Service Repository') {
+                steps {
+                    script {
+                        utils.createRepo("${TAVROS_GIT_PROVIDER}")
+                    }
+                }
+            }
+            stage('Setup Project') {
+                steps {
+                    container('maven') {
+                        script {
+                            utils.writeResource "camelwebservice.jenkinsfile", "Jenkinsfile"
```

Comment:
we can wrap the writeResource with dir so that we don't have to do the sh move. no functional gain, but it'd mirror what the openapi quickstart is doing iirc

Reply:
I tried this initially and spent awhile trying to debug it, but I got an AccessDeniedException when I wrapping it with `dir`. I thought it might be because it was created in a different container? which is why I tried to wrap this in the maven container too, but that didn't help (I will remove that).

I don't know if there's anything sensitive in the stacktrace so I'll message it to you.